node_api/tests/unit/logic/IMQtoSQL.spec.ts

```diff
@@ -0,0 +1,246 @@
+import { expect, test } from "vitest";
+import { IMQtoSQL } from "@/logic/IMQtoSQL";
+import { Query } from "@im-library/interfaces/AutoGen";
+import { server } from "../../setupTests";
+import EntityService from "@/services/entity.service";
+import axios from "axios";
+import { IM } from "@im-library/vocabulary";
+test("IMQtoSQL", async () => {
+  server.close();
+
+  // const svc = new EntityService(axios);
+  // const entity = await svc.getPartialEntity("http://endhealth.info/im#Q_TestQuery", [IM.DEFINITION]);
+  // const entity = await svc.getPartialEntity("urn:uuid:7d9a0a98-4df8-4748-8940-061a5148293c", [IM.DEFINITION]); // APL Search
+  // const entity = await svc.getPartialEntity("urn:uuid:6d517466-813b-46a8-b848-aaf5a4fbdcbf", [IM.DEFINITION]); // SMI Population
+  // const entity = await svc.getPartialEntity("urn:uuid:40a4a1f1-b768-4db8-a8a6-6df744935d97", [IM.DEFINITION]); // SMI Priority 1
+
+  // let json = entity.data[IM.DEFINITION];
+  // const def: Query = JSON.parse(json);
+
+  const def: any = {
```

Comment:
same query is used in multiple tests, maybe we could look in to adding it to a separate file